### PR TITLE
lsp: Return copies of cache maps

### PR DESCRIPTION
**Pull request by charlieegan3**
Fixes: https://github.com/StyraInc/regal/issues/750

### CODE REVIEW

File: internal/lsp/cache/cache.go [hunk 1]
```diff
@@ -84,7 +84,12 @@ func (c *Cache) GetAllFiles() map[string]string {
 	c.fileContentsMu.Lock()
 	defer c.fileContentsMu.Unlock()
 
-	return c.fileContents
+	cp := make(map[string]string, len(c.fileContents))
```
Comment (srenatus) [inline internal/lsp/cache/cache.go:87]: I believe we could use [`maps.Clone`](https://pkg.go.dev/golang.org/x/exp/maps#Clone) for this

**Review by anderseknert — APPROVED** [review]
LGTM, but @srenatus suggestion sounds good

File: internal/lsp/cache/cache.go (hunk 1 above)
Comment (charlieegan3) [reply internal/lsp/cache/cache.go]: Ohh go is cool now.

Comment (charlieegan3) [reply internal/lsp/cache/cache.go]: I have added this in 51758077f96655a00f3ec60bc68e0d5f98fceac3

**Review by anderseknert — APPROVED** [review]

**Pull request merged**